From 62a18d6eb72ca95f91dee190b9eaa44d6c5557ee Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Tue, 20 Apr 2021 13:14:06 +0200
Subject: [PATCH] kABI: cover up change in struct kvm_arch

References: bsc#1184969
Patch-mainline: never, kABI

struct kvm_arch should not be used by out-of-tree drivers.

Signed-off-by: Michal Suchanek <msuchanek@suse.de>
---
 arch/powerpc/include/asm/kvm_host.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/powerpc/include/asm/kvm_host.h b/arch/powerpc/include/asm/kvm_host.h
index e6e5f59aaa97..c74241d205f5 100644
--- a/arch/powerpc/include/asm/kvm_host.h
+++ b/arch/powerpc/include/asm/kvm_host.h
@@ -299,7 +299,11 @@ struct kvm_arch {
 	struct list_head spapr_tce_tables;
 	struct list_head rtas_tokens;
 	struct mutex rtas_token_lock;
+#ifndef __GENKSYMS__
 	DECLARE_BITMAP(enabled_hcalls, MAX_HCALL_OPCODE/4 + 1);
+#else
+	DECLARE_BITMAP(enabled_hcalls, H_SCM_PERFORMANCE_STATS/4 + 1);
+#endif
 #endif
 #ifdef CONFIG_KVM_MPIC
 	struct openpic *mpic;
-- 
2.26.2

